prometheus_juju_backup_all_exporter/core.py

```diff
@@ -0,0 +1,126 @@
+"""Module for collecter core codes."""
+
+from abc import abstractmethod
+from dataclasses import dataclass
+from logging import getLogger
+from typing import Dict, Iterable, List, Type
+
+from prometheus_client.metrics_core import Metric
+from prometheus_client.registry import Collector
+
+from .config import Config
+
+logger = getLogger(__name__)
+
+
+@dataclass
+class Payload:
+    """Container of data for each timeseries."""
+
+    name: str
+    value: float
+    labels: List[str]
+    uuid: str = ""  # timeseries's name
+
+    def __post_init__(self) -> None:
+        """Create uuid based on metric name and labels."""
+        self.uuid = f"{self.name}({self.labels})"
+
+
+@dataclass
+class Specification:
+    """Specification for metrics."""
+
+    name: str
+    labels: List[str]
+    documentation: str
+    metric_class: Type[Metric]
+
+
+class SyncCollector(Collector):
```

Comment:
Just a suggestion: I think name `BlockingCollector` might be better to emphasize nature of this collector.

Reply:
Updated